### PR TITLE
Re-arrange modules for clarity

### PR DESCRIPTION
**Pull request by ahoppen**
This resolves the FIXMEs that I had in `Documentation/Modules.md` and generally improves the cohesiveness of modules.

### CODE REVIEW

**Comment by ahoppen** [comment]
@swift-ci Please test

**Comment by ahoppen** [comment]
@swift-ci Please test Windows

**Comment by ahoppen** [comment]
@swift-ci Please test Windows

**Comment by ahoppen** [comment]
@swift-ci Please test Windows

**Comment by ahoppen** [comment]
@swift-ci Please test Windows

File: Documentation/Modules.md [hunk 1]
```diff
@@ -8,6 +8,10 @@ Swift types to represent the [Build Server Protocol (BSP) specification, version
 
 FIXME: Add link for BSP and version
 
+### BuildSystemIntegration
+
+Defines the queries SourceKit-LSP can ask of a a build system, like getting compiler arguments for a file. Finding a target’s dependencies or preparing a target.
```
Comment (plemarquand) [inline Documentation/Modules.md:13]:
```suggestion
Defines the queries SourceKit-LSP can ask of a build system, like getting compiler arguments for a file, finding a target’s dependencies or preparing a target.
```

File: Documentation/Modules.md [hunk 2]
```diff
@@ -36,69 +40,27 @@ Swift types to represent the [Language Server Protocol (LSP) specification, vers
...
-- MainFilesProvider.swift
-- PathPrefixMapping.swift
-- SplitShellCommand.swift
-- WorkspaceType.swift
-
-#### ToolchainRegistry
+### SKLogging
 
-Discovers Swift toolchains on the system.
+Types that are API-compatible with OSLog to allow logging to OSLog when building for Darwin platforms and logging to stderr or files on non-Darwin platforms. This should not be dependent on any LSP specific types and be portable to other packages.
```
Comment (plemarquand) [inline Documentation/Modules.md:51]:
```suggestion
Types that are API-compatible with OSLog that allow logging to OSLog when building for Darwin platforms and logging to stderr or files on non-Darwin platforms. This should not be dependent on any LSP specific types and be portable to other packages.
```

**Comment by ahoppen** [comment]
@swift-ci Please test

**Review by bnbarham — APPROVED** [review]

**Comment by ahoppen** [comment]
@swift-ci Please test

**Comment by ahoppen** [comment]
@swift-ci Please test Windows

**Pull request merged**